### ISSUE TITLE
About test results

### ISSUE THREAD

**Issue by wefwefWEF2** [cchen-cc/MA-SAM#10]
Hi， thanks for the great job.
When I reproduced the results of the paper, I found that the result of label 12 was always 0 and it was difficult to reach the result of vit b of 82.5. I did not change any code. Is it related to batchsize? 

python test.py --adapt_ckpt /code/medical/MA-SAM_12_28/results/epoch_279.pth --data_path /data/dyn123/medical_image/MA-SAM/synapseCT/Training/2D_all_5slice --ckpt /code/medical/Transseg/src/backbones/encoders/pretrained_models/sam_vit_b_01ec64.pth --is_savenii  --vit_name vit_b 
Namespace(adapt_ckpt='/code/medical/MA-SAM_12_28/results/epoch_279.pth', data_path='/data/dyn123/medical_image/MA-SAM/synapseCT/Training/2D_all_5slice', num_classes=12, img_size=512, seed=1234, is_savenii=True, deterministic=1, ckpt='/code/medical/Transseg/src/backbones/encoders/pretrained_models/sam_vit_b_01ec64.pth', vit_name='vit_b', rank=32, scale=1.0, module='sam_fact_tt_image_encoder', output_dir='/code/medical/MA-SAM_12_28/results/epoch_279')
  0%|                                                                                                                                                                                 | 0/6 [00:00<?, ?it/s]idx 1 case 0035 mean_dice 0.738734
 17%|████████████████████████████▏                                                                                                                                            | 1/6 [00:39<03:19, 39.87s/it]idx 1 case 0036 mean_dice 0.668650
 33%|████████████████████████████████████████████████████████▎                                                                                                                | 2/6 [02:01<04:18, 64.51s/it]idx 1 case 0037 mean_dice 0.675733
 50%|████████████████████████████████████████████████████████████████████████████████████▌                                                                                    | 3/6 [02:42<02:41, 53.86s/it]idx 1 case 0038 mean_dice 0.678954
 67%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋                                                        | 4/6 [03:25<01:38, 49.36s/it]idx 1 case 0039 mean_dice 0.688280
 83%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊                            | 5/6 [04:03<00:45, 45.24s/it]idx 1 case 0040 mean_dice 0.671483
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6/6 [05:25<00:00, 54.29s/it]
Mean class 1 name spleen mean_dice 0.937284
Mean class 2 name right kidney mean_dice 0.914729
Mean class 3 name left kidney mean_dice 0.927163
Mean class 4 name gallbladder mean_dice 0.166667
Mean class 5 name esophagus mean_dice 0.666901
Mean class 6 name liver mean_dice 0.954785
Mean class 7 name stomach mean_dice 0.729955
Mean class 8 name aorta mean_dice 0.855800
Mean class 9 name vena mean_dice 0.780666
Mean class 10 name vein mean_dice 0.622591
Mean class 11 name pancreas mean_dice 0.687127
Mean class 12 name adrenal gland mean_dice 0.000000
Testing performance in best val model: mean_dice : 0.686972

**Comment by cchen-cc** [comment]
Did you change the batch size?

**Comment by wefwefWEF2** [comment]
> 您更改了批量大小吗？

Yes, I have checked it is related with batch size, thanks for your reply.
For box prompt in your code, could you please provide some prepare prompt code for this dataloader?

 prompt_pth = self.masks_list[idx].replace('/masks/2Dmask_', '/boxprompts/2Dprompt_')
            boxprompt = read_image(prompt_pth)

**Comment by MaoFuyou** [comment]
你好 我也是测试了 159pth 在 btcv数据上！但是 result 结果比较离谱

<img width="1043" alt="Image" src="https://github.com/user-attachments/assets/114e11f7-2ba6-42bb-99f6-830a664a8894" />

不知道 应该从哪里排查

**Comment by mrbird643** [comment]
> Hi， thanks for the great job. When I reproduced the results of the paper, I found that the result of label 12 was always 0 and it was difficult to reach the result of vit b of 82.5. I did not change any code. Is it related to batchsize?
> 
> python test.py --adapt_ckpt /code/medical/MA-SAM_12_28/results/epoch_279.pth --data_path /data/dyn123/medical_image/MA-SAM/synapseCT/Training/2D_all_5slice --ckpt /code/medical/Transseg/src/backbones/encoders/pretrained_models/sam_vit_b_01ec64.pth --is_savenii --vit_name vit_b Namespace(adapt_ckpt='/code/medical/MA-SAM_12_28/results/epoch_279.pth', data_path='/data/dyn123/medical_image/MA-SAM/synapseCT/Training/2D_all_5slice', num_classes=12, img_size=512, seed=1234, is_savenii=True, deterministic=1, ckpt='/code/medical/Transseg/src/backbones/encoders/pretrained_models/sam_vit_b_01ec64.pth', vit_name='vit_b', rank=32, scale=1.0, module='sam_fact_tt_image_encoder', output_dir='/code/medical/MA-SAM_12_28/results/epoch_279') 0%| | 0/6 [00:00<?, ?it/s]idx 1 case 0035 mean_dice 0.738734 17%|████████████████████████████▏ | 1/6 [00:39<03:19, 39.87s/it]idx 1 case 0036 mean_dice 0.668650 33%|████████████████████████████████████████████████████████▎ | 2/6 [02:01<04:18, 64.51s/it]idx 1 case 0037 mean_dice 0.675733 50%|████████████████████████████████████████████████████████████████████████████████████▌ | 3/6 [02:42<02:41, 53.86s/it]idx 1 case 0038 mean_dice 0.678954 67%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████▋ | 4/6 [03:25<01:38, 49.36s/it]idx 1 case 0039 mean_dice 0.688280 83%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▊ | 5/6 [04:03<00:45, 45.24s/it]idx 1 case 0040 mean_dice 0.671483 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6/6 [05:25<00:00, 54.29s/it] Mean class 1 name spleen mean_dice 0.937284 Mean class 2 name right kidney mean_dice 0.914729 Mean class 3 name left kidney mean_dice 0.927163 Mean class 4 name gallbladder mean_dice 0.166667 Mean class 5 name esophagus mean_dice 0.666901 Mean class 6 name liver mean_dice 0.954785 Mean class 7 name stomach mean_dice 0.729955 Mean class 8 name aorta mean_dice 0.855800 Mean class 9 name vena mean_dice 0.780666 Mean class 10 name vein mean_dice 0.622591 Mean class 11 name pancreas mean_dice 0.687127 Mean class 12 name adrenal gland mean_dice 0.000000 Testing performance in best val model: mean_dice : 0.686972

Could your replication be close to the results presented in the paper?
